### PR TITLE
refactor!: Inconsistent naming of Edgex commands

### LINKED ISSUE #173: Inconsistent naming of Onvif functions to EdgeX commands
Currently the service has inconsistent names for EdgeX commands compared to the onvif equivalent. For example, some commands remove the `Get` such as the `StreamUri` command, however others such as `GetPresets` keep the `Get` in the beginning.

Here is a list of all the EdgeX commands which begin with `Get` or `Set`:

```
SetSystemFactoryDefault
GetVideoEncoderConfigurations
GetNodes
GetNode
GetConfigurations
GetConfigurationOptions
GetStatus
SetPreset
GetPresets
SetHomePosition
GetEventProperties
GetAnalyticsConfigurations
GetSupportedAnalyticsModules
GetAnalyticsModuleOptions
GetSupportedRules
GetRuleOptions
```

Obtained via the command:
```shell
cat cmd/res/profiles/camera.yaml | sed -En 's/.*name: "((Get|Set)[^"]+)".*/\1/p'
```

-------

Here is a scraped table with all of the EdgeX commands and their Onvif equivalents. The difference
field is what you get if you remove the edgex command name from the onvif call name:

| REST | EdgeX Command | Onvif Call | Difference |
| ---------|-------------------------|---------------|-------------- |
| GET | Hostname | GetHostname | Get |
| PUT | Hostname | SetHostname | Set |
| GET | DNS | GetDNS | Get |
| PUT | DNS | SetDNS | Set |
| GET | NetworkInterfaces | GetNetworkInterfaces | Get |
| PUT | NetworkInterfaces | SetNetworkInterfaces | Set |
| GET | NetworkProtocols | GetNetworkProtocols | Get |
| PUT | NetworkProtocols | SetNetworkProtocols | Set |
| GET | NetworkDefaultGateway | GetNetworkDefaultGateway | Get |
| PUT | NetworkDefaultGateway | SetNetworkDefaultGateway | Set |
| GET | DeviceInformation | GetDeviceInformation | Get |
| GET | SystemDateAndTime | GetSystemDateAndTime | Get |
| PUT | SystemDateAndTime | SetSystemDateAndTime | Set |
| PUT | SetSystemFactoryDefault | SetSystemFactoryDefault |  |
| PUT | SystemReboot | SystemReboot |  |
| GET | MetadataConfigurations | GetMetadataConfigurations | Get |
| GET | MetadataConfiguration | GetMetadataConfiguration | Get |
| PUT | MetadataConfiguration | SetMetadataConfiguration | Set |
| GET | CompatibleMetadataConfigurations | GetCompatibleMetadataConfigurations | Get |
| GET | MetadataConfigurationOptions | GetMetadataConfigurationOptions | Get |
| PUT | AddMetadataConfiguration | AddMetadataConfiguration |  |
| PUT | RemoveMetadataConfiguration | RemoveMetadataConfiguration |  |
| GET | Profiles | GetProfiles | Get |
| GET | StreamUri | GetStreamUri | Get |
| GET | GetVideoEncoderConfigurations | GetVideoEncoderConfigurations |  |
| GET | VideoEncoderConfiguration | GetVideoEncoderConfiguration | Get |
| PUT | VideoEncoderConfiguration | SetVideoEncoderConfiguration | Set |
| GET | VideoEncoderConfigurationOptions | GetVideoEncoderConfigurationOptions | Get |
| GET | Users | GetUsers | Get |
| PUT | Users | SetUser | SetUser |
| PUT | CreateUsers | CreateUsers |  |
| PUT | DeleteUsers | DeleteUsers |  |
| GET | DiscoveryMode | GetDiscoveryMode | Get |
| PUT | DiscoveryMode | SetDiscoveryMode | Set |
| PUT | AddScopes | AddScopes |  |
| GET | Scopes | GetScopes | Get |
| PUT | Scopes | SetScopes | Set |
| PUT | RemoveScopes | RemoveScopes |  |
| GET | GetNodes | GetNodes |  |
| GET | GetNode | GetNode |  |
| GET | GetConfigurations | GetConfigurations |  |
| GET | Configuration | GetConfiguration | Get |
| PUT | Configuration | SetConfiguration | Set |
| GET | GetConfigurationOptions | GetConfigurationOptions |  |
| PUT | AddPTZConfiguration | AddPTZConfiguration |  |
| PUT | RemovePTZConfiguration | RemovePTZConfiguration |  |
| PUT | AbsoluteMove | AbsoluteMove |  |
| PUT | RelativeMove | RelativeMove |  |
| PUT | ContinuousMove | ContinuousMove |  |
| GET | GetStatus | GetStatus |  |
| PUT | Stop | Stop |  |
| PUT | SetPreset | SetPreset |  |
| GET | GetPresets | GetPresets |  |
| PUT | GotoPreset | GotoPreset |  |
| PUT | RemovePreset | RemovePreset |  |
| PUT | GotoHomePosition | GotoHomePosition |  |
| PUT | SetHomePosition | SetHomePosition |  |
| PUT | SendAuxiliaryCommand | SendAuxiliaryCommand |  |
| GET | GetEventProperties | GetEventProperties |  |
| GET | Media2Profiles | GetProfiles | GetProfiles |
| GET | GetAnalyticsConfigurations | GetAnalyticsConfigurations |  |
| PUT | AddConfiguration | AddConfiguration |  |
| PUT | RemoveConfiguration | RemoveConfiguration |  |
| GET | GetSupportedAnalyticsModules | GetSupportedAnalyticsModules |  |
| GET | AnalyticsModules | GetAnalyticsModules | Get |
| PUT | AnalyticsModules | ModifyAnalyticsModules | Modify |
| PUT | CreateAnalyticsModules | CreateAnalyticsModules |  |
| PUT | DeleteAnalyticsModules | DeleteAnalyticsModules |  |
| GET | GetAnalyticsModuleOptions | GetAnalyticsModuleOptions |  |
| GET | GetSupportedRules | GetSupportedRules |  |
| GET | Rules | GetRules | Get |
| PUT | Rules | ModifyRules | Modify |
| GET | GetRuleOptions | GetRuleOptions |  |
| PUT | CreateRules | CreateRules |  |
| PUT | DeleteRules | DeleteRules |  |

### PR DESCRIPTION
**Pull request by presatish**
BREAKING CHANGE: Making Edgex command names consistent by removing `GET` and `SET` keywords.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [X] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [X] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

## Testing Instructions
make test
make docker
/compose-builder$ git checkout main
/compose-builder$ make pull
/compose-builder$ make gen no-secty ds-onvif-camera
Replace onvif service docker image in docker-compose.yml with your latest image created using make docker from the previous step
/compose-builder$ make up
/bin/configure-subnets.sh without error
/bin/map-credentials.sh without error
Onvif devices should be registered in Edgex
Get the updated postman collection and env from the doc folder, update the env with the names of onvif cameras and run the collection without error.

### CODE REVIEW

File: doc/postman/onvif.postman_collection.json
Comment (ajcasagrande) [inline doc/postman/onvif.postman_collection.json]: this file uses raw SOAP onvif messages and should not be modified. my mistake.

**Review by ajcasagrande — COMMENTED** [review]
The changes look good. I will hold off on Accepting it until I further check the non-modified stuff to see if I spot any missed items.

File: cmd/res/profiles/camera.yaml
Comment (ChristianDarr-personal) [inline cmd/res/profiles/camera.yaml:363]: Should this be named PTZConfigurations? Some of the other configuration commands include the service that its configuring.

File: cmd/res/profiles/camera.yaml
Comment (ChristianDarr-personal) [inline cmd/res/profiles/camera.yaml:382]: Same as above.

File: doc/openapi/README.md
Comment (ChristianDarr-personal) [inline doc/openapi/README.md:4]: Should these point to the files on github or the url to the docs site itself?

**Review by ChristianDarr-personal — COMMENTED** [review]
Looks pretty good. I just had a few small suggestions, nothing critical though

File: cmd/res/profiles/camera.yaml
Comment (presatish) [reply cmd/res/profiles/camera.yaml:363]: I think it should be fine as it is consistent with the ONVIF function name `GetConfigurations`

File: cmd/res/profiles/camera.yaml
Comment (presatish) [reply cmd/res/profiles/camera.yaml:382]: Same reply as above.

File: doc/openapi/README.md
Comment (presatish) [reply doc/openapi/README.md:4]: You mean should they point to files in this repo? But were they not all moved to Edgex docs?

Comment (ChristianDarr-personal) [reply doc/openapi/README.md:4]: I mean using this link instead: https://docs.edgexfoundry.org/3.0/microservices/device/supported/device-onvif-camera/supplementary-info/ONVIF-protocol/#tested-onvif-cameras

Comment (ajcasagrande) [reply doc/openapi/README.md:4]: The reason I pointed them to github is because it helps the developer know where to find the file that needs to be modified.

File: cmd/res/profiles/camera.yaml
Comment (ajcasagrande) [reply cmd/res/profiles/camera.yaml:363]: I think I kind of agree with @ChristianDarr-personal here, as the prefix could help avoid confusion. The thing you have to remember is that the onvif APIs have separate scoped endpoints, so `GetConfigurations` is calling `GetConfigurations` of the `ptz` scope, so that is why the name does not include it. In EdgeX we do not have a concept of scope, all commands are in the same area. The command `DeviceCapabilities` is calling the `GetCapabilities` of the `device` scope. Similarily, the `MediaCapabilities` calls `GetCapabilities` of the `media` scope. without that distinction, the name would be clashing.

Comment (ajcasagrande) [reply cmd/res/profiles/camera.yaml:363]: It could be argued that all of the endpoints begin with the service name to be consistant and explicit, but it seems like overkill in certain situations. In other cases (especially PTZ) i think it would help avoid confusion. `Node`, `Nodes`, `Presets`, are good examples of that.

Comment (presatish) [reply cmd/res/profiles/camera.yaml:363]: All the other ONVIF configuration related functions have the scope names included in them such as `GetMetadataConfiguration`, `GetVideoEncoderConfiguration`, etc.. except this. I left it from consistency perspective. I can change into `PTZConfigurations` if that's going to help, the others I don't think there is a need. Let me know.

**Review by ajcasagrande — DISMISSED** [review]

**Comment by lenny-goodell** [comment]
I defer to @ajcasagrande & @ChristianDarr-personal on this one. 

You do need to fix the commit message and DCO issues

**Comment by lenny-goodell** [comment]
@presatish BTW, the commit message should be a `refactor!:` not a `fix:` , this is not a bug, but a tech debt and it is a BREAKING CHNAGE. Please update accordingly.

**Review by lenny-goodell — APPROVED** [review]

**Pull request merged**
